src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,129 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementArray;
+    private int size;
+    private int capacity;
+
+    public ArrayList() {
+        capacity = DEFAULT_CAPACITY;
+        elementArray = (T[]) new Object[capacity];
+    }
+
+    public T[] elementArrayPlusSize(T[] elementArray, int increasingInSizeByInt) {
```

Comment:
All methods that are used only in the class must be private
All private methods should be below public methods
```suggestion
    private T[] grow(T[] elementArray, int increasingInSizeByInt) {
```